packages/cli/src/runners/task-runner-module.ts

```diff
@@ -0,0 +1,80 @@
+import { TaskRunnersConfig } from '@n8n/config';
+import * as a from 'node:assert/strict';
+import Container, { Service } from 'typedi';
+
+import type { TaskRunnerProcess } from '@/runners/task-runner-process';
+
+import { TaskRunnerWsServer } from './runner-ws-server';
+import type { LocalTaskManager } from './task-managers/local-task-manager';
+import type { TaskRunnerServer } from './task-runner-server';
+
+/**
+ * Module responsible for loading and starting task runner. Task runner can be
+ * run either internally (=launched by n8n as a child process) or externally
+ * (=launched by some other orchestrator)
+ */
+@Service()
+export class TaskRunnerModule {
+	private taskRunnerHttpServer: TaskRunnerServer | undefined;
+
+	private taskRunnerWsServer: TaskRunnerWsServer | undefined;
+
+	private taskManager: LocalTaskManager | undefined;
+
+	private taskRunnerProcess: TaskRunnerProcess | undefined;
+
+	constructor(private readonly runnerConfig: TaskRunnersConfig) {}
+
+	async start() {
+		a.ok(!this.runnerConfig.disabled, 'Task runner is disabled');
+
+		await this.loadTaskManager();
+		await this.loadTaskRunnerServer();
+
+		if (this.runnerConfig.mode !== 'external') {
+			await this.startInternalTaskRunner();
+		}
+	}
+
+	async stop() {
```

Comment:
Looks like this is unused? Should this have an `@OnShutdown` decorator?

Reply:
It's currently used only for tests. My idea is to make the module responsible for shutting down its individual components, but at the moment it's still the individual components that have the `@OnShutdown` decorator. I'll do that as a followup